src/services/withPath.ts

```diff
@@ -0,0 +1,13 @@
+import { isUrl, Url } from '@/types/url'
+
+export function withPath(url: Url, path: string): Url
+export function withPath(url: string, path: string): Url
+export function withPath(url: string, path: string): Url {
```

Comment:
This accepts a `url` but it assumes that the url doesn't have a query or hash. Should these utilities account for urls of different shapes? Right now they're order dependent, which is probably fine as long as they're only internal utilities.